### DISCUSSION TITLE
Feedback & Plans for Lavalink v4

### DISCUSSION THREAD

**Discussion by topi314** [lavalink-devs/Lavalink#807] [Q&A]
Hey Lavalink people,
With Lavalink `v3.7.0` finally being released new and the new REST API being in your hands, I'm here to share our plans moving forward with Lavalink `v4`.
We don't have any big breaking changes for you, but we are looking into one often requested feature. We are planning on adding support for `artworkUrl` & `isrc` in all tracks. Unfortunately this requires breaking changes to the current encoded track format and therefore some clients will require updating their implementation to work with `v4`. 
But you don't worry if you have a large collection of previously encoded tracks as they will still work just fine, they just won't have an `artworkUrl` or `isrc`.

Here is a full list of all the changes we are currently planning to do for `v4`:
* Removal of deprecated endpoints in `v3.7.0` (all non `/v3` prefixed endpoints except `/version`)
* Replacing the `resumeKey` with the `sessionId` to resume a Lavalink session
* Removal of all json fields which were deprecated in `v3` (`track` in events, etc.)
* Addition of `artworkUrl` and `isrc` in the track json & encoded tracks
* Addition to the plugin api to be able modify the track json. See https://github.com/freyacodes/Lavalink/pull/788 for more info
* Addition of the full track json in all `TrackXEvent`s
* Refactor of the `LoadResult` json

If you have any thoughts or ideas about these changes or something else you'd like to see please let us know here as we depend on your feedback.

**Comment by flav-code** [comment]
Add an endpoint to see all client connected to the node with WS

Return an array

agent
bot id

**Reply by freyacodes** [reply]
I don't see that being a particularly useful feature for most people. It could be implemented as a plugin fairly easily though.

**Reply by flav-code** [reply]
making a plugin just for that, it's not worth it and I find it useful to see the active websocket sessions

**Reply by topi314** [reply]
One Lavalink session doesn't need to know about any other Lavalink Session, they are isolated from each other. There is no point in adding this.

And additional plugin could provide more than just that single endpoint like a full dashboard to see all connections & even disconnect individual ones

**Reply by flav-code** [reply]
ok, could you make this plugin when you have some time ?

**Reply by topi314** [reply]
no
